### DISCUSSION TITLE
No API command for new "Done" state of notifications

### DISCUSSION THREAD

**Discussion by ltguillaume** [community/community#24653] [Apps, API and Webhooks]
<p>It seems there is no API command to either read or set the new “Done” (completed) triage feature for notifications. As such, third-party applications cannot use the API to clean the inbox.</p>
<p>I have searched high and low for either documentation or anything mentioning this, but for now this issue seems to be completely discarded.</p>

**Comment by francisfuzz** [comment]
<p>:wave: Welcome to the GitHub Support Community, <a class="mention" href="/ltguillaume">@ltGuillaume</a>!</p>
<aside class="quote no-group" data-username="ltGuillaume" data-post="1" data-topic="122508" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 ltGuillaume:</div>
<blockquote>
<p>It seems there is no API command to either read or set the new “Done” (completed) triage feature for notifications. As such, third-party applications cannot use the API to clean the inbox.</p>
<p>I have searched high and low for either documentation or anything mentioning this, but for now this issue seems to be completely discarded.</p>
</blockquote>
</aside>
<p>We can confirm here that there’s no API endpoint for achieving that goal. We’re always working to improve GitHub and the GitHub Support Community, and we consider every suggestion we receive.</p>

**Comment by ltguillaume** [comment]
<p>Thanks for your reply. I’ve submitted a request as you suggested.</p>

**Comment by vchirikov** [comment]
<p>8 months have passed and there is no api for that.<br>
<a class="mention" href="/francisfuzz">@francisfuzz</a> any news?</p>

**Comment by francisfuzz** [comment]
<p>:wave: <a class="mention" href="/vchirikov">@vchirikov</a> - I’m not aware of any new REST API endpoints or GraphQL API mutations for accomplishing this nor am I aware of any planned feature work for this. I wish I had better news to share, though, if it helps for keeping up with the latest changes, I suggest monitoring <a href="https://docs.github.com/en/graphql/overview/breaking-changes">Breaking changes</a> and our <a href="https://docs.github.com/en/graphql/overview/changelog">GraphQL Changelog</a> for the latest.</p>

**Comment by xorima** [comment]
<aside class="quote group-github-employees" data-username="francisfuzz" data-post="2" data-topic="122508">
<div class="title">
<div class="quote-controls"></div>
 francisfuzz:</div>
<blockquote>
<p>We can confirm here that there’s no API endpoint for achieving that goal. We’re always working to improve GitHub and the GitHub Support Community, and we consider every suggestion we receive.</p>
</blockquote>
</aside>
<p>Still waiting for this here. Very frustrating that a developer first company isn’t API first when building things…</p>

**Comment by jsumners** [comment]
<p>I suppose they don’t want us to automate getting rid of the stress in our lives.</p>

**Comment by tkapin** [comment]
@francisfuzz - Almost two years passed since your last reply, could you please share an update on this? Thanks in advance!

**Reply by francisfuzz** [reply]
@tkapin - Hi there! I no longer work on this team nor am I able to provide any updates since my last reply two years ago. With the move over to using GitHub Discussions, I believe there are product managers that monitor these threads, but I cannot speak on their behalf or on upcoming features, but it may be worth checking out the GitHub public roadmap:

https://github.com/orgs/github/projects/4247

I hope that helps.

**Reply by tkapin** [reply]
Understood, thanks for your reply and the link!

**Reply by ltguillaume** [reply]
There's nothing on the roadmap concerning this, as far as I can see.

**Reply by tkapin** [reply]
Agree, I also haven't seen anything on the roadmap that would indicate this API will be provided.

**Comment by loic-sharma** [comment]
GitHub has an undocumented `markNotificationAsDone` GraphQL mutation, however, it seems "regular" personal access tokens do not have permission to use this mutation.

Example request:

```http
POST https://api.github.com/graphql HTTP/1.1
# Headers omitted

{"operationName":"MarkNotificationAsDone","query":"mutation MarkNotificationAsDone($id: ID!) {\n  markNotificationAsDone(input: {id: $id}) {\n    __typename\n    success\n  }\n}","variables":{"id":"NT_ABCDEFG"}}
```

It'd be nice if notification GraphQL queries/mutations were made publicly available. I created a similar thread to allow saving notifications through the API too: https://github.com/community/community/discussions/39606

**Reply by tkapin** [reply]
Awesome, this is useful, thank you.

**Comment by clareliguori** [comment]
I figured out you can do this with the `gh` CLI:

```
gh api graphql -f query='mutation { markNotificationAsDone(input: {id: "ID_HERE"}) { success } }'
```

**Comment by jharvey10** [comment]
I'd built a custom notifications app relying on these graphql notification queries+mutations which worked amazingly... for about two weeks until they were removed from the graphql endpoint (at least from my account's perspective).

AFAIK there's currently no suitable alternative, and to make things worse, when you visit a PR page (for example), it appears to automatically mark the relevant notification as "read", which messes with any logic relying on `/notifications?all=false` because they just disappear from the results.

Back to the drawing board 😖 The gql APIs are so nice I wish we could just have them.

**Comment by clareliguori** [comment]
I also saw it disappeared from the GraphQL API.  This is working for me now:
```
IDS=$(gh api --paginate '/notifications' --jq '.[] | select(.reason == "approval_requested") | .id')
echo "$IDS" | xargs -P 20 -I {} gh api -X DELETE '/notifications/threads/{}'
```
